src/wazuh_testing/modules/api/utils.py

```diff
@@ -92,21 +92,24 @@ def login(user: str = WAZUH_API_USER, password: str = WAZUH_API_PASSWORD,
         response (requests.Response): Response object.
 
     Raises:
-        RuntimeError(msg, requests.Response): When could not login after `login_attempts` every `sleep_time`
+        RuntimeError(msg, requests.Response): When could not login after `login_attempts` every timeout determined by 
+        the `backoff_factor`.
     """
     url = f"{get_base_url(protocol=protocol, host=host, port=port)}{LOGIN_ROUTE}"
 
-    for _ in range(login_attempts):
-        response = requests.post(url, headers=set_authorization_header(user, password), verify=WAZUH_API_CERTIFICATE, timeout=timeout)
+    session = requests.Session()
+    retry = Retry(total=None, connect=login_attempts, backoff_factor=backoff_factor)
```

Comment:
`total` is the `Total number of retries to allow`, while `connect` is `How many connection-related errors to retry on` ([ref](https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.Retry)). 

In my opinion it would be better to use `total`, is there any reason to use `connect` instead? We want it to be retried `login_attempts` times, regardless of the error type.

```suggestion
    retry = Retry(total=login_attempts, backoff_factor=backoff_factor)
```

Reply:
The reason is that it would block rate limit errors and the ones that are expected to occur. This may cause the rate limit to expire and thus the tests calls to pass. Strangely, this would happen even if I set it to `0` (I'm not sure if it's doing one retry there or not, but in the tests I ran it caused trouble) and setting it to `None` would use the default value.

Some tests are checking that the API is indeed rejecting more connections (test_DOS_blocking_system.py, test_bruteforce_blocking_system.py), so this way we just retry on connection-related issues